### ISSUE TITLE
Zip files containing all the notebooks not available on Windows

### ISSUE THREAD

**Issue by lesteve** [sphinx-gallery/sphinx-gallery#183]
From https://github.com/sphinx-gallery/sphinx-gallery/pull/182#issuecomment-266415318

 The zip files to download all the examples of the gallery don't seem to be available, a backslash is missing in the path. Problem in py2 and py3
```
C:\projects\sphinx-gallery\doc\auto_examples\index.rst:279: WARNING: download file not readable: C:\projects\sphinx-gallery\doc\auto_examplesauto_examples_python.zip
C:\projects\sphinx-gallery\doc\auto_examples\index.rst:285: WARNING: download file not readable: C:\projects\sphinx-gallery\doc\auto_examplesauto_examples_jupyter.zip
C:\projects\sphinx-gallery\doc\index.rst:64: WARNING: toctree contains reference to nonexisting document 'auto_mayavi_examples/index'
C:\projects\sphinx-gallery\doc\tutorials\index.rst:87: WARNING: download file not readable: C:\projects\sphinx-gallery\doc\tutorialstutorials_python.zip
C:\projects\sphinx-gallery\doc\tutorials\index.rst:93: WARNING: download file not readable: C:\projects\sphinx-gallery\doc\tutorialstutorials_jupyter.zip
```

This is an excerpt of the generated file doc/auto_examples/index.rst:
```
  .. container:: sphx-glr-download

    :download:`Download all examples in Jupyter notebooks: tutorials_jupyter.zip </tutorials\tutorials_jupyter.zip>`
```

I guess this issue is a good opportunity to review the use of backslashes vs forward slashes in sphinx-gallery.

**Comment by lesteve** [comment]
> I guess this issue is a good opportunity to review the use of backslashes vs forward slashes in sphinx-gallery.

My take on this is to use forward slashes as much as possible even on Windows since they work for files. I am not sure what the best practices are on Windows. An alternative is to always normalise the path maybe via `os.path.normpath`.

**Comment by jrleeman** [comment]
We're running up against this with MetPy doc builds as well. Any more thinking on how this could be tackled? Maybe even using `os.sep` when creating the paths. We might have time to dig in with a few pointers at some point, but it looks like some relatively big changes coming up in #160

**Comment by lesteve** [comment]
I would be in favour of merging a small fix for this one before #160 so a PR would be very welcome indeed.

Having said that I have not looked at this problem in detail so I am not sure how this could be tackled. The questions that come to mind and that may help you get started
In this excerpt of doc/auto_examples/index.rst
```
  .. container:: sphx-glr-download

    :download:`Download all examples in Jupyter notebooks: tutorials_jupyter.zip </tutorials\tutorials_jupyter.zip>`
```
* is it fine only using backslashes for the link target
* why do we have a mix of forward and backslash in the link target
* should we be more careful in sphinx-gallery about making the difference between files (where you want to use `os.sep`) and url parts (where you always want to use forward-slash irrespective of the OS)
